vela-templates/definitions/internal/workflowstep/build-source-code.cue

```diff
@@ -0,0 +1,120 @@
+import (
+	"vela/op"
+)
+
+"build-source-code": {
```

Comment:
We currently support the step called `build-image` which uses kaniko to build images, maybe you can add the pvc capability in that step instead of creating a new one?

Reply:
OK, I'll try to make some adaptions.

Reply:
In that way, how can we get the generated build artifacts like the `yarn build` output files? Do we need kaniko's capability to build container images?

I've tried to deploy a app using step `build-push-image` and component `webservice` per [this guide](https://kubevela.net/docs/next/tutorials/s2i), and the process is simple and smooth enough. Users probably will rarely have such a need to persist files for later extra upload...

Reply:
Hmmmm, your step will only build the code but not the image, right? Maybe it should be separated with the step `build-push-image`, but a general step for code building is hard since there're multiple languages, I'm not sure if there're some mature tools in the community that we can combie with.

Reply:
- I found that kaniko can also specify the arg like `--destination=/output` to persist the generated files in volume.
- If only need to build code, maybe using a container is enough. The user provides a dockerfile to specify the base image and build commands in steps.
- For multiple langs support, no suitable tools were found yet. I found that most of them are cicd platforms.

I currently think of two possible options for code building, one is user provides a `dockerfile`, or user provides the base `image` and build `cmd` which is the current draft implementation.